src/Options.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Equip\Command;
+
+/**
+ * A general purpose value object for command options.
+ *
+ * When constructed, all required options must be passed.
+ *
+ * @since 1.3.0
+ */
+abstract class Options
```

Comment:
No immutable mutators (i.e. `with*()` methods)? If not, that's fine, just curious.